pkg/download/buffered_reader.go

```diff
@@ -82,8 +87,12 @@ func (p *readerPool) Get() *bufferedReader {
 func (p *readerPool) Put(br *bufferedReader) {
 	br.c.L.Lock()
 	defer br.c.L.Unlock()
+	if br.pool == nil {
+		return
+	}
 	br.ready = false
 	br.err = nil
+	br.pool = nil
 	br.buf.Reset()
```

Comment:
Thinking this through, these likely should be moved up to .Close() and not be part of .Put().